lib/ansible/modules/notification/cisco_spark.py

```diff
@@ -0,0 +1,205 @@
+    require: true
+    aliases: ['token']
+
+  message:
+    description:
+      - The message you would like to send.
+    required: True
+'''
+
+EXAMPLES = """
+# Note: The following examples assume a variable file has been imported
+# that contains the appropriate information.
+
+- name: Cisco Spark - Markdown Message to a Room
+  cisco_spark:
+    recipient_type: roomId
+    recipient_id: "{{ room_id }}"
+    message_type: markdown
+    personal_token: "{{ token }}"
+    message: "**Cisco Spark Ansible Module - Room Message in Markdown**"
+
+- name: Cisco Spark - Text Message to a Room
+  cisco_spark:
+    recipient_type: roomId
+    recipient_id: "{{ room_id }}"
+    message_type: text
+    personal_token: "{{ token }}"
+    message: "Cisco Spark Ansible Module - Room Message in Text"
+
+- name: Cisco Spark - Text Message by an Individuals ID
+  cisco_spark:
+    recipient_type: toPersonId
+    recipient_id: "{{ person_id}}"
+    message_type: text
+    personal_token: "{{ token }}"
+    message: "Cisco Spark Ansible Module - Text Message to Individual by ID"
+
+- name: Cisco Spark - Text Message by an Individuals E-Mail Address
+  cisco_spark:
+    recipient_type: toPersonEmail
+    recipient_id: "{{ person_email }}"
+    message_type: text
+    personal_token: "{{ token }}"
+    message: "Cisco Spark Ansible Module - Text Message to Individual by E-Mail"
+
+"""
+
+RETURN = """
+status_code:
+  description:
+    - The Response Code returned by the Spark API
+    - Full Responsde Code explanations can be found at https://developer.ciscospark.com/endpoint-messages-post.html
+  returned: always
+  type: int
+  sample: 200
+
+message:
+    description:
+      - The Response Message returned by the Spark API
+      - Full Responsde Code explanations can be found at https://developer.ciscospark.com/endpoint-messages-post.html
+    returned: always
+    type: string
+    sample: OK (585 bytes)
+"""
+
+
+def spark_message(module):
+    """ When check mode is specified, establish a read only connection, that does not return any user specific
+    data, to validate connectivity. In regular mode, send a message to a Cisco Spark Room or Individual"""
+
+    # Ansible Specific Variables
+    results = {}
+    ansible = module.params
+
+    headers = {
+        'Authorization': 'Bearer {}'.format(ansible['personal_token']),
+        'content-type': 'application/json'
+    }
+
+    if module.check_mode:
+
+        url = "https://api.ciscospark.com/v1/people/me"
+
+        response, info = fetch_url(module, url, headers=headers)
+
+    else:
+
+        url = "https://api.ciscospark.com/v1/messages"
+
+        # By Room ID
+        payload = {
+            ansible['recipient_type']: ansible['recipient_id'],
+            ansible['message_type']: ansible['message']
+        }
+
+        payload = module.jsonify(payload)
+
+        response, info = fetch_url(module, url, data=payload, headers=headers)
```

Comment:
I extremely appreciate your patience here and I understand the approach you are trying to take. I understand your frustration and all I ask is that you bare with me while we find an approach that works. 

I fully understand the goal of not duplicating code. That is why I deleted the duplicate header and various status_code variables as much as possible. I should have seen that approach sooner and I thank you for pushing back on removing the function I had.

That being said, there was a reason why I had two separate fetch_url calls. In my testing I would receive a error response code from the API when I included a payload for the authentication code. I confirmed that this time around again but then decided to check one last time. I can't explain the difference between the code but it is not working and I will be submitting a new commit that solves the issue.